phoenicis-javafx/src/main/java/org/phoenicis/javafx/views/mainwindow/ui/SearchBox.java

```diff
@@ -1,31 +1,34 @@
 package org.phoenicis.javafx.views.mainwindow.ui;
 
-import javafx.event.ActionEvent;
-import javafx.event.EventHandler;
-import org.phoenicis.javafx.views.common.ThemeManager;
-
 import javafx.beans.property.StringProperty;
-import javafx.scene.control.TextField;
 import javafx.scene.control.Button;
+import javafx.scene.control.TextField;
 import javafx.scene.layout.AnchorPane;
 
 import java.util.function.Consumer;
 
 public class SearchBox extends AnchorPane {
     private TextField searchField;
     private Button clearButton;
+    private Consumer<String> onSearch;
```

Comment:
I thought about this. But I'm not sure. `Optional` always implies for me that it's not needed. I do not want to give that impression. What I did shall only be error handling to avoid exceptions but it's not really a valid way to use the `SearchBox`. Maybe I should even write some error log if `onSearch` is `null`.